leigharobinson/dementia_music_memory
# wrote code for patient_song viewset

# Description
In View, I added create, list, retrieve definitions to deal with the many to many relationship patien_song table 
Fixes # (ability to access on the frontend props of Patient and Song)
## Type of change
Please delete options that are not relevant.
# Testing Instructions
In Postman, Create a Caretaker/User by going to http://127.0.0.1:8000/register/ and make a POST request using 

{
        "username": "TestUser",
        "email":"TestUser123@gmail.com",
        "password" :"123",
        "first_name": "Test",
        "last_name" : "Testerson",
        "title": "Nurse"
}
<img width="587" alt="Screen Shot 2020-09-12 at 10 37 54 PM" src="https://user-images.githubusercontent.com/58920011/93009593-9d13ea80-f548-11ea-993f-5d67c1632bcf.png">

Take the Token and add it to Headers 
<img width="820" alt="Screen Shot 2020-09-12 at 10 39 10 PM" src="https://user-images.githubusercontent.com/58920011/93009621-c896d500-f548-11ea-8cf2-3f5e7f5d7b23.png">

Then make a patient. 
Go to http://127.0.0.1:8000/patients and make a POST request.
In the Body add this information:


{
    "first_name": "Bobby",
    "last_name": "Bobberson",
    "diagnosis": "Alzheimers",
    "year_of_birth": "1925",
    "facility_id": 1
}

if you need to remember the id your Caretaker got assigned on register, make a GET request to http://127.0.0.1:8000/caretakers


Then go to http://127.0.0.1:8000/patientssongs and select POST, in the Body place this dummy data:

{
    "song_id": 1,
    "patient_id": 1
}
Then you can make a GET request to http://127.0.0.1:8000/patientssongs and make sure the POST went through


# Checklist:
